go.mod

```diff
@@ -204,7 +204,7 @@ require (
 	github.com/libp2p/go-libp2p-noise v0.2.0 // indirect
 	github.com/libp2p/go-libp2p-peerstore v0.2.8 // indirect
 	github.com/libp2p/go-libp2p-pnet v0.2.0 // indirect
-	github.com/libp2p/go-libp2p-pubsub v0.4.2 // indirect
+	github.com/libp2p/go-libp2p-pubsub v0.4.2 // direct
```

Comment:
should we try to update ipfs and libp2p first, before reviewing and testing this? there have been two "major" versions of libp2p-pubsub and related libraries since v0.4, so I worry that we'd have to re-do the review and testing when we inevitably upgrade.

it seems easiest to first upgrade libp2p in master, which should be relatively easy, then rebase this on top to then merge and test.

@p4u thoughts?